sources/ClangSharp/Extensions/CXModuleMapDescriptor.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+
+namespace ClangSharp
+{
+    public unsafe partial struct CXModuleMapDescriptor : IDisposable
+    {
+        public static CXModuleMapDescriptor Create(uint options)
+        {
+            return clang.ModuleMapDescriptor_create(options);
+        }
+
+        public void Dispose()
+        {
+            if (Pointer != IntPtr.Zero)
+            {
+                clang.ModuleMapDescriptor_dispose(this);
+                Pointer = IntPtr.Zero;
+            }
+        }
+
+        public CXErrorCode SetFrameworkModuleName(string name)
+        {
+            using (var marshaledName = new MarshaledString(name))
```

Comment:
This is how I opted to do the string marshaling (and cleanup) currently.

Ideally the "safe" API would take a `UTF8String` rather than `string`, but that type doesn't exist yet (and `ReadOnlySpan<byte>` support is fairly limited in `netstandard2.0`).